gen/__init__.py

```diff
@@ -524,20 +524,9 @@ def build_late_package(late_files, config_id, provider):
     }
 
 
-def generate(
-        arguments,
-        extra_templates=list(),
-        extra_sources=list(),
-        extra_targets=list()):
-    # To maintain the old API where we passed arguments rather than the new name.
-    user_arguments = arguments
-    arguments = None
-
-    sources, targets, templates = get_dcosconfig_source_target_and_templates(
-        user_arguments, extra_templates, extra_sources)
-
+def validate_and_get_arguments(sources, targets):
```

Comment:
This function is doing a few things (resolving/validating config, collecting argument dict, collecting late variables) that seem like they could be split up. Collecting late variables is a clear candidate for a separate function -- that this function returns two unrelated values is awkward.